Foundation/NSCompoundPredicate.swift

```diff
@@ -44,11 +44,17 @@ public class NSCompoundPredicate : NSPredicate {
     }
 
     override public func evaluateWithObject(object: AnyObject?, substitutionVariables bindings: [String : AnyObject]?) -> Bool {
+        let subpredicates = self.subpredicates as! [NSPredicate]
+        
         switch compoundPredicateType {
         case .AndPredicateType:
-            return subpredicates.reduce(true, combine: { $0 && $1.evaluateWithObject(object, substitutionVariables: bindings) })
+            return subpredicates.reduce(true, combine: {
+                $0 && $1.evaluateWithObject(object, substitutionVariables: bindings)
+            })
         case .OrPredicateType:
-            return subpredicates.reduce(false, combine: { $0 || $1.evaluateWithObject(object, substitutionVariables: bindings) })
+            return subpredicates.reduce(false, combine: {
+                $0 || $1.evaluateWithObject(object, substitutionVariables: bindings)
+            })
```

Comment:
I don't see how that's true. I added test cases for both "and" and "or predicates based on your example code here, and I did not get any test failures or crashes. The && and || operators short circuit properly on their own, and I don't see how reduce() could alter that behavior.